Content.Client/Stack/StackVisualizerComponent.cs

```diff
@@ -0,0 +1,72 @@
+using Robust.Shared.Utility;
+
+namespace Content.Client.Stack;
+
+/// <summary>
+/// Visualizer for items that come in stacks and have different appearance
+/// depending on the size of the stack. Visualizer can work by switching between different
+/// icons in <c>_spriteLayers</c> or if the sprite layers are supposed to be composed as transparent layers.
+/// The former behavior is default and the latter behavior can be defined in prototypes.
+///
+/// <example>
+/// <para>To define a Stack Visualizer prototype insert the following
+/// snippet (you can skip Appearance if already defined)
+/// </para>
+/// <code>
+/// - components:
+///   - type: StackVisualizer
+///     stackLayers:
+///       - goldbar_10
+///       - goldbar_20
+///       - goldbar_30
+/// </code>
+/// </example>
+/// <example>
+/// <para>Defining a stack visualizer with composable transparent layers</para>
+/// <code>
+/// - components:
+///   - type: StackVisualizer
+///     composite: true
+///     stackLayers:
+///       - cigarette_1
+///       - cigarette_2
+///       - cigarette_3
+///       - cigarette_4
+///       - cigarette_5
+///       - cigarette_6
+/// </code>
+/// </example>
+///  <seealso cref="_spriteLayers"/>
+/// </summary>
+[RegisterComponent]
+[Access(typeof(StackVisualizerSystem))]
+public sealed class StackVisualizerComponent : Component
```

Comment:
move to stack component (also VV fields and format attributes above the fields, not in the same line)

Reply:
The logic is also used by `ItemCounterComponent`. Do you want me to copypasta to its system as well?